components/layout/inline.rs

```diff
@@ -1069,7 +1069,21 @@ impl InlineFlow {
             for style in fragment.inline_styles() {
                 match style.get_box().vertical_align {
                     vertical_align::T::baseline => {}
-                    vertical_align::T::middle => {}
+                    vertical_align::T::middle => {
+                        // This is easier to understand if you think of it as:
+                        //
+                        //    block_start = line_block_metrics.start +
+                        //      line_block_metrics.size / 2 -
+                        //      (fragment_inline_metrics.ascent +
+                        //       fragment_inline_metrics.depth_below_baseline) / 2
+                        //
+                        // (That is, subtract out the pre-existing `block_start` value.)
+                        block_start = block_start -
+                            line_block_metrics.size_above_baseline +
+                            line_block_metrics.size.scale_by(0.5) +
+                            fragment_inline_metrics.ascent.scale_by(0.5) -
+                            fragment_inline_metrics.depth_below_baseline.scale_by(0.5)
```

Comment:
Should this be `fragment.border_box.size.block.scale_by(0.5)` to find the middle of the fragment?  Currently if the fragment is an `<img>` whose height is different from the line height, it doesn't align correctly.